sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSenderExtensions.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Search.Documents.Models;
+
+namespace Azure.Search.Documents
+{
+    /// <summary>
+    /// Extension methods for <see cref="SearchIndexingBufferedSender{T}"/>.
+    /// </summary>
+    public static class SearchIndexingBufferedSenderExtensions
+    {
+        /// <summary>
+        /// Adds a batch of delete actions to eventually send to the search
+        /// index.
+        /// </summary>
+        /// <param name="indexer">The sender.</param>
+        /// <param name="key">The name of the key field.</param>
+        /// <param name="documentKeys">
+        /// The keys of the documents to delete.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be canceled.
+        /// </param>
+        public static void DeleteDocuments(
```

Comment:
Why aren't these defined on the sender? I don't see what value (yet?) making them extension methods provides.

Reply:
These only work in the dynamic `SearchDocument` case and `Sender<T>` won't let me specialize.